Add initial page describing the OME security process (#48)
ome/ome-contributing: docs/security-process.rst

Security process
================

> OME makes extensive use of GitHub  for managing vulnerability reports and fixes - see

Reviewer: ```suggestion
OME makes extensive use of GitHub for managing vulnerability reports and fixes - see
```

> https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities.

Reviewer: Maybe clarify here that this document is for OME maintainers, not the process for external contributors to report a security issue?

If the reported problem is not identified as a security risk and a draft advisory has
been created, it must be closed with a comment explaining why it is not considered.

Fix a reported vulnerability
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If the problem has been identified as a security risk, a temporary private fork of
the relevant repository should be created from the draft advisory, see
https://docs.github.com/en/code-security/tutorials/fix-reported-vulnerabilities/collaborate-in-a-fork/.

Collaborators can be added and removed from the private fork as described in 
https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities/fix-reported-vulnerabilities/add-collaborators
and https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities/fix-reported-vulnerabilities/remove-collaborators
as necessary.

The development and review process can then proceed as usual using pull requests
against the private fork.

Announce the security release
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Once a fix has been identified, a pre-announcement of the upcoming security release
should be made on the image.sc forum including at least the `ome` and `security` tags.

> The delay between the publication pre-announcement and the security release  may vary

Reviewer: ```suggestion
The delay between the publication pre-announcement and the security release may vary
```

depending on a number of factors. For OMERO security releases, sufficient notice should
be given to system administrators to plan and schedule a downtime and upgrade.

Prepare the advisory
^^^^^^^^^^^^^^^^^^^^

Prior to its publication, the content advisory should be updated with:

- the package, affected versions and patched versions
- a description including Background, Impact, Workaround and Resolution
- an assessment of the severity using CVSS v3 base metrics
- a CVE, if an identifier has already been assigned as part of the vulnerability report,
  it should be re-used, otherwise, a CVE should be requested from the GitHub advisory
- credits to the reporter if applicable

> Publish the advisory

Reviewer: Maybe link this section to the release process pages as well? Then we have a more obvious reminder on release days.

Author: Agreed with cross-linking from the individual release process pages. Would it be best to replace the existing `Register CVE` section (as a security process should likely be considered before you start entering the release steps? And/or also as an optional publication step?

Reviewer: I think something like we have at the top of https://ome-contributing.readthedocs.io/en/latest/bioformats-release-process.html, that clearly states `Before starting the release process...` and then links to the new security page. I don't have strong feelings about whether that should be its own section on the release page as for `Register CVE`.

Author: [889488d](https://github.com/ome/ome-contributing/pull/48/commits/889488de0424850b09db2f8dc0597dfe7835c3cb) adds an introductory paragraph and unifies the styling of the Bio-Formats, OMERO.server and OMERO.web release process pages

^^^^^^^^^^^^^^^^^^^^

> On release data, all open Pull Requests against the temporary fork should be merged

Reviewer: ```suggestion
On release day, all open Pull Requests against the temporary fork should be merged
```

in the security advisory - see https://docs.github.com/en/code-security/tutorials/fix-reported-vulnerabilities/collaborate-in-a-fork#merging-changes-in-a-security-advisory
Once merged, all changes will be visible on the public repository.

The release process can then follow the standard procedures with the following variations:

- the GitHub advisory should be published as described in
  https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities/fix-reported-vulnerabilities/publish-repository-advisory.

> - i naddition to the release announcement, the GitHub advisory should be duplicated to the

Reviewer: ```suggestion
- in addition to the release announcement, the GitHub advisory should be duplicated to the
```

Security process
================

This document describes the security process for OME maintainers. The process for
reporting a security vulnerability is described in the
`security page <https://www.openmicroscopy.org/security/>`_ of the OME website.
This process makes extensive use of GitHub functionalities for managing vulnerability
reports and fixes - see
https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities.

> 

Reviewer 2: Should we add another section here about responding to the reporter via email by the next workday, outlining the general process and timeline in the email, to avoid them reporting the issue elsewhere?

Create draft advisory
^^^^^^^^^^^^^^^^^^^^^

> Once a vulnerability is [reported](https://www.openmicroscopy.org/security/) through

Reviewer: ```suggestion
Once a vulnerability is `reported <https://www.openmicroscopy.org/security/>`_ through
```

Sorry, one more suggestion since the formatting of this link on https://ome-contributing--48.org.readthedocs.build/en/48/security-process.html is a bit off.

Author: Thanks! Ah the Markdown / reStructuredText dance...

the relevant repository should be created from the draft advisory, see
https://docs.github.com/en/code-security/tutorials/fix-reported-vulnerabilities/collaborate-in-a-fork/.

Collaborators can be added and removed from the private fork as described in 
https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities/fix-reported-vulnerabilities/add-collaborators
and https://docs.github.com/en/code-security/how-tos/report-and-fix-vulnerabilities/fix-reported-vulnerabilities/remove-collaborators
as necessary.

The development and review process can then proceed as usual using pull requests
against the private fork.

Announce the security release
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Once a fix has been identified, a pre-announcement of the upcoming security release
should be made on the image.sc forum including at least the `ome` and `security` tags.
The delay between the publication pre-announcement and the security release may vary
depending on a number of factors. For OMERO security releases, sufficient notice should
be given to system administrators to plan and schedule a downtime and upgrade.

> 

Reviewer 2: Also add a link to the image.sc post in the `section.announcement` on the website here: https://github.com/ome/www.openmicroscopy.org/blob/c4369f300e75be1bed44a707c621eece2dea0230/index.html#L44
